apps/server/src/core/logger/logger.service.ts

```diff
@@ -1,42 +1,65 @@
-import { ConsoleLogger, Injectable, LogLevel, Scope } from '@nestjs/common';
-import { ConfigService } from '@nestjs/config';
+import { Inject, Injectable, Scope } from '@nestjs/common';
+import { WINSTON_MODULE_PROVIDER } from 'nest-winston';
 import util from 'util';
-import { ILoggerConfig, RequestLoggingBody } from './interfaces';
-import { AvailableLogLevel, ILogger } from './interfaces/logger.interface';
+import { Logger as WinstonLogger } from 'winston';
+import { RequestLoggingBody } from './interfaces';
+import { ILogger } from './interfaces/logger.interface';
 
 @Injectable({ scope: Scope.TRANSIENT })
 /**
  * Default logger for server application.
  * Must implement ILogger but must not extend ConsoleLogger (this can be changed).
  * Transient injection: Wherever injected, a separate instance will be created, that can be provided with a custom context.
  */
-export class Logger extends ConsoleLogger implements ILogger {
+export class Logger implements ILogger {
 	/**
 	 * This Logger Service can be injected into every Class,
 	 * use setContext() with CustomProviderClass.name that will be added to every log.
 	 * It implements @ILogger which provides the logger methods.
 	 * CAUTION: PREPARE STRINGS AS LOG DATA, DO NOT LOG COMPLEX DATA STRUCTURES
 	 */
-	constructor(private readonly configService: ConfigService<ILoggerConfig, true>) {
-		super();
-		const logLevels = this.configService.get<AvailableLogLevel[]>('AVAILABLE_LOG_LEVELS');
-		this.setLogLevels(logLevels as LogLevel[]);
+	private context = '';
+
+	constructor(@Inject(WINSTON_MODULE_PROVIDER) private readonly logger: WinstonLogger) {}
+
+	log(message: unknown, context?: string | undefined): void {
+		this.logger.log('info', this.createMessage(message, context));
+	}
+
+	warn(message: unknown, context?: string | undefined): void {
+		this.logger.warning(this.createMessage(message, context));
+	}
+
+	debug(message: unknown, context?: string | undefined): void {
+		this.logger.debug(this.createMessage(message, context));
+	}
+
+	verbose(message: unknown, context?: string | undefined): void {
+		this.logger.verbose(this.createMessage(message, context));
 	}
 
 	http(message: RequestLoggingBody, context?: string): void {
-		const logLevel = 'http';
-		if (!this.isLevelEnabled(logLevel as LogLevel)) {
-			return;
-		}
-		this.printMessages([JSON.stringify(message)], context || this.context, 'HTTP Request' as LogLevel);
+		this.logger.http(this.createMessage(message, context));
```

Comment:
please change this to logger.notice.

The function should stay for backwards compability, it should just write to notice instead.

Reply:
✔️